coffea/processor/executor.py

```diff
@@ -774,18 +773,38 @@ def localsource(path):
                     except KeyError:
                         # emit warning if not found?
                         pass
-                df = NanoEvents.from_file(
-                    file=file,
-                    treename=item.treename,
-                    entrystart=item.entrystart,
-                    entrystop=item.entrystop,
-                    metadata={
-                        'dataset': item.dataset,
-                        'filename': item.filename
-                    },
-                    cache=cache,
-                )
+                if issubclass(schema, schemas.BaseSchema):
+                    file = uproot.open(item.filename)
+                    factory = NanoEventsFactory.from_file(
+                        file=file,
+                        treepath=item.treename,
+                        entry_start=item.entrystart,
+                        entry_stop=item.entrystop,
+                        runtime_cache=cache,
+                        schemaclass=schema,
+                        metadata={
+                            'dataset': item.dataset,
+                            'filename': item.filename,
+                        },
+                    )
+                    df = factory.events()
+                elif issubclass(schema, NanoEvents):
+                    file = uproot.open(item.filename, localsource=localsource, xrootdsource=xrootdsource)
+                    df = NanoEvents.from_file(
```

Comment:
For awkward0 NanoEvents I'm pretty sure you need to pass an uproot3 file object for it to work. How about we just pass the filename to `NanoEvents` so that it can open with uproot3 and then we never import `uproot3` inside this module. You can set `file = None` and skip the later checks and performance counters when this path is taken.

Reply:
Since the `uproot3` module is used in other places like:
```python
            from uproot.source.xrootd import XRootDSource
            xrootdsource = XRootDSource.defaults
            xrootdsource['timeout'] = xrootdtimeout
```
or
```python
    if mmap:
        localsource = {}
    else:
        opts = dict(uproot.FileSource.defaults)
        opts.update({'parallel': None})

        def localsource(path):
            return uproot.FileSource(path, **opts)
```
we're going to have to import `uproot3` anyway. That is, at least until #350 is merged in. Do you still want me to make this change?

Reply:
You're right, in fact now I realize that the awkward1/uproot4 version is the only one not working here.